### DISCUSSION TITLE
sudo command not found

### DISCUSSION THREAD

**Discussion by sunknudsen** [AppImage/AppImageKit#1366] [Q&A]
I created a VeraCrypt AppImage using `appimagetool-x86_64.AppImage` following a [guide](https://github.com/veracrypt/VeraCrypt/issues/501#issuecomment-554319569) found on VeraCrypt's GitHub repo.

Everything works as expected but I run into an [issue](https://github.com/veracrypt/VeraCrypt/issues/501#issuecomment-802796035) whey trying to mount encrypted volumes on Tails (likely this is not an AppImage issue, but just in case, reaching out here).

> Failed to obtain administrative privileges: sudo /tmp/.mount_Ver********/usr/bin/veracrypt: command not found

Would someone happen to know why `sudo` is not found? Is there anything that can be done at the AppImage creation level?

```console
$ which sudo
/usr/bin/sudo
```

**Comment by azubieta** [comment]
@sunknudsen Some system may restrict which files can be executed as sudo. I can imagine that a mounted path will provably fall under this restriction in a security hardened system.

**Comment by sunknudsen** [comment]
@azubieta Naïve question... running VeraCrypt AppImage as root (which is likely a bad idea) works... does this reveal anything we could use to troubleshoot the issue?

Is it possible AppImage doesn't have `/usr/bin/` in its `PATH`?

**Comment by azubieta** [comment]
> does this reveal anything we could use to troubleshoot the issue?

Maybe you could share the AppImage in question for analysis 

> Is it possible AppImage doesn't have /usr/bin/ in its PATH?

It's possible, but in the log you shared it "seems" to be using a full path.

**Comment by sunknudsen** [comment]
> Maybe you could share the AppImage in question for analysis

@azubieta Sure, here it is including PGP signature.

[veracrypt-1.24-Update7.zip](https://github.com/AppImage/AppImageKit/files/6172699/veracrypt-1.24-Update7.zip)

Thanks for helping out!

**Comment by sunknudsen** [comment]
> It's possible, but in the log you shared it "seems" to be using a full path.

@azubieta I meant that `/usr/bin` would not be in `PATH` environment variable so `sudo` is not found.

**Comment by probonopd** [comment]
Is this applicable here?
https://github.com/AppImage/AppImageKit/issues/146#issuecomment-220702575

In any case, what happens if you run `sudo <your AppImage>`?

**Comment by sunknudsen** [comment]
@probonopd Thanks for helping out... running AppImage using sudo works although file navigator (in VeraCrypt) doesn't include some paths (such as external flash drives) included when running as unprivileged user on Tails (in other words, UX is a little broken, but I am able to mount encrypted volume).

**Comment by probonopd** [comment]
Try `sudo -E ...` to preserve the environment variables.
